From 9b9fd2a0d9ca81aa16ddfe2f7e219b94e2ac158b Mon Sep 17 00:00:00 2001
From: Andrew Bartlett <abartlet@samba.org>
Date: Mon, 16 Aug 2021 14:46:31 +1200
Subject: [PATCH] mit-kdc: Remove build time support for KDB_API < 10

The previous commits restricted to MIT KDC build to MIT 1.19 and this removes the
 #ifdef in the code of what will become untested code.

Signed-off-by: Andrew Bartlett <abartlet@samba.org>
Reviewed-by: Andreas Schneider <asn@samba.org>

Autobuild-User(master): Andreas Schneider <asn@cryptomilk.org>
Autobuild-Date(master): Thu Aug 26 07:05:44 UTC 2021 on sn-devel-184
---
 source4/kdc/mit-kdb/kdb_samba.h            | 32 ------------------
 source4/kdc/mit-kdb/kdb_samba_policies.c   | 38 ----------------------
 source4/kdc/mit-kdb/kdb_samba_principals.c |  7 ----
 3 files changed, 77 deletions(-)

diff --git a/source4/kdc/mit-kdb/kdb_samba.h b/source4/kdc/mit-kdb/kdb_samba.h
index ad4f6e275739..8a29334bcea5 100644
--- a/source4/kdc/mit-kdb/kdb_samba.h
+++ b/source4/kdc/mit-kdb/kdb_samba.h
@@ -71,18 +71,11 @@ krb5_error_code kdb_samba_db_put_principal(krb5_context context,
 krb5_error_code kdb_samba_db_delete_principal(krb5_context context,
 					      krb5_const_principal princ);
 
-#if KRB5_KDB_API_VERSION >= 8
 krb5_error_code kdb_samba_db_iterate(krb5_context context,
 				     char *match_entry,
 				     int (*func)(krb5_pointer, krb5_db_entry *),
 				     krb5_pointer func_arg,
 				     krb5_flags iterflags);
-#else
-krb5_error_code kdb_samba_db_iterate(krb5_context context,
-				     char *match_entry,
-				     int (*func)(krb5_pointer, krb5_db_entry *),
-				     krb5_pointer func_arg);
-#endif
 
 /* from kdb_samba_masterkey.c */
 
@@ -114,21 +107,6 @@ krb5_error_code kdb_samba_dbekd_encrypt_key_data(krb5_context context,
 
 /* from kdb_samba_policies.c */
 
-#if KRB5_KDB_API_VERSION < 10
-krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
-					    unsigned int flags,
-					    krb5_const_principal client_princ,
-					    krb5_db_entry *client,
-					    krb5_db_entry *server,
-					    krb5_db_entry *krbtgt,
-					    krb5_keyblock *client_key,
-					    krb5_keyblock *server_key,
-					    krb5_keyblock *krbtgt_key,
-					    krb5_keyblock *session_key,
-					    krb5_timestamp authtime,
-					    krb5_authdata **tgt_auth_data,
-					    krb5_authdata ***signed_auth_data);
-#else
 krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
 					    unsigned int flags,
 					    krb5_const_principal client_princ,
@@ -147,7 +125,6 @@ krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
 					    void *authdata_info,
 					    krb5_data ***auth_indicators,
 					    krb5_authdata ***signed_auth_data);
-#endif
 
 krb5_error_code kdb_samba_db_check_policy_as(krb5_context context,
 					     krb5_kdc_req *kdcreq,
@@ -162,7 +139,6 @@ krb5_error_code kdb_samba_db_check_allowed_to_delegate(krb5_context context,
 						       const krb5_db_entry *server,
 						       krb5_const_principal proxy);
 
-#if KRB5_KDB_API_VERSION >= 9
 void kdb_samba_db_audit_as_req(krb5_context kcontext,
 			       krb5_kdc_req *request,
 			       const krb5_address *local_addr,
@@ -171,14 +147,6 @@ void kdb_samba_db_audit_as_req(krb5_context kcontext,
 			       krb5_db_entry *server,
 			       krb5_timestamp authtime,
 			       krb5_error_code error_code);
-#else
-void kdb_samba_db_audit_as_req(krb5_context kcontext,
-			       krb5_kdc_req *request,
-			       krb5_db_entry *client,
-			       krb5_db_entry *server,
-			       krb5_timestamp authtime,
-			       krb5_error_code error_code);
-#endif
 
 /* from kdb_samba_change_pwd.c */
 
diff --git a/source4/kdc/mit-kdb/kdb_samba_policies.c b/source4/kdc/mit-kdb/kdb_samba_policies.c
index c431567a7f41..ac9865aac601 100644
--- a/source4/kdc/mit-kdb/kdb_samba_policies.c
+++ b/source4/kdc/mit-kdb/kdb_samba_policies.c
@@ -290,21 +290,6 @@ static krb5_error_code ks_verify_pac(krb5_context context,
 	return code;
 }
 
-#if KRB5_KDB_API_VERSION < 10
-krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
-					    unsigned int flags,
-					    krb5_const_principal client_princ,
-					    krb5_db_entry *client,
-					    krb5_db_entry *server,
-					    krb5_db_entry *krbtgt,
-					    krb5_keyblock *client_key,
-					    krb5_keyblock *server_key,
-					    krb5_keyblock *krbtgt_key,
-					    krb5_keyblock *session_key,
-					    krb5_timestamp authtime,
-					    krb5_authdata **tgt_auth_data,
-					    krb5_authdata ***signed_auth_data)
-#else
 krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
 					    unsigned int flags,
 					    krb5_const_principal client_princ,
@@ -323,7 +308,6 @@ krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
 					    void *authdata_info,
 					    krb5_data ***auth_indicators,
 					    krb5_authdata ***signed_auth_data)
-#endif
 {
 	krb5_authdata **authdata = NULL;
 	krb5_boolean is_as_req;
@@ -331,10 +315,8 @@ krb5_error_code kdb_samba_db_sign_auth_data(krb5_context context,
 	krb5_pac pac = NULL;
 	krb5_data pac_data;
 
-#if KRB5_KDB_API_VERSION >= 10
 	krbtgt = krbtgt == NULL ? local_krbtgt : krbtgt;
 	krbtgt_key = krbtgt_key == NULL ? local_krbtgt_key : krbtgt_key;
-#endif
 
 	/* FIXME: We don't support S4U yet */
 	if (flags & KRB5_KDB_FLAGS_S4U) {
@@ -477,7 +459,6 @@ static void samba_bad_password_count(krb5_db_entry *client,
 	}
 }
 
-#if KRB5_KDB_API_VERSION >= 9
 void kdb_samba_db_audit_as_req(krb5_context context,
 			       krb5_kdc_req *request,
 			       const krb5_address *local_addr,
@@ -499,22 +480,3 @@ void kdb_samba_db_audit_as_req(krb5_context context,
 
 	/* TODO: perform proper audit logging for addresses */
 }
-#else
-void kdb_samba_db_audit_as_req(krb5_context context,
-			       krb5_kdc_req *request,
-			       krb5_db_entry *client,
-			       krb5_db_entry *server,
-			       krb5_timestamp authtime,
-			       krb5_error_code error_code)
-{
-	/*
-	 * FIXME: This segfaulted with a FAST test
-	 * FIND_FAST: <unknown client> for <unknown server>, Unknown FAST armor type 0
-	 */
-	if (client == NULL) {
-		return;
-	}
-
-	samba_bad_password_count(client, error_code);
-}
-#endif
diff --git a/source4/kdc/mit-kdb/kdb_samba_principals.c b/source4/kdc/mit-kdb/kdb_samba_principals.c
index c17fe8b0a14d..a8c99b025c9a 100644
--- a/source4/kdc/mit-kdb/kdb_samba_principals.c
+++ b/source4/kdc/mit-kdb/kdb_samba_principals.c
@@ -311,18 +311,11 @@ krb5_error_code kdb_samba_db_delete_principal(krb5_context context,
 	return KRB5_KDB_DB_INUSE;
 }
 
-#if KRB5_KDB_API_VERSION >= 8
 krb5_error_code kdb_samba_db_iterate(krb5_context context,
 				     char *match_entry,
 				     int (*func)(krb5_pointer, krb5_db_entry *),
 				     krb5_pointer func_arg,
 				     krb5_flags iterflags)
-#else
-krb5_error_code kdb_samba_db_iterate(krb5_context context,
-				     char *match_entry,
-				     int (*func)(krb5_pointer, krb5_db_entry *),
-				     krb5_pointer func_arg)
-#endif
 {
 	struct mit_samba_context *mit_ctx;
 	krb5_db_entry *kentry = NULL;
